src/Sulu/Component/Content/Mapper/Translation/MultipleTranslatedProperties.php

```diff
@@ -49,8 +49,10 @@ public function __construct(
         $this->languageNamespace = $languageNamespace;
         $this->properties = [];
         foreach ($names as $name) {
-            $propertyName = (!empty($namespace) ? $namespace . '-' : '') . $name;
-            $this->properties[$name] = new Property($propertyName, [], 'none', false, true);
+            if (empty($this->properties[$name])) {
```

Comment:
Any idea why this might be called multiple times? :confused:

Reply:
I've had a deeper look at this, and it feels like you are fixing this in the wrong place. I don't know when exactly you are entering this (for me it was when saving a page int he backend), but when I was having a look at this, it was not being executed multiple times for each name... But from what I have seen, the `$names` array being iterated here contained the value of `created` twice. That's the only name you would skip with this implementation. However, it would make more sense to find out why there are double entries in the `$names` array.

Reply:
More context: When I was saving the page, the `ContentQueryBuilder` was called, and the `created` property is listed twice [here](https://github.com/sulu/sulu/blob/develop/src/Sulu/Component/Content/Query/ContentQueryBuilder.php#L56). Does your performance also improve when you if you remove the doubled entry? Or do you have a completely different entry point to this constructor?

Reply:
Also, when the language is switched, it looks to me like the [`ExcerptStructureExtension`](https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/ContentBundle/Content/Structure/ExcerptStructureExtension.php#L177) adds the properties without clearing it before. It feels like these are the issues we should address. And if we address all of them, we should have the same perfomance improvement as if we do it this way.

Reply:
As wrote above those lines are not the problem but more like a late cleanup. Still in my opinion there should be a check because it could happen all the time. If you don't want a check that just ignores the duplication maybe we should here raise an exception?

Otherwise you would never ever know that this happens and only an import of many nodes will really run into that problem.

Reply:
Raising an exception actually sounds like a good idea. Then it should also be quite easy to fix all these issues now, shouldn't it? The exception will exactly tell you where you need to fix stuff. Otherwise the duplicated properties might make other problems as well :confused: